### DISCUSSION TITLE
My Gas Counter integration

### DISCUSSION THREAD

**Discussion by turkelali** [arendst/Tasmota#18086] [Q&A]
Hello,

I want to interface my Gas counter values to my homeassistant through tasmota. Gas counter have optical sensor with IEC 62056-21 standart and I have optical reader/writer like: https://www.ebay.com/itm/354359025520
My gas counter is https://zennergas.eu/wp-content/uploads/sites/6/2022/01/2.Blocker_EN_ZRG_BK0Z_WG25S-Rev05.pdf
Currently I am only getting /?! in console and tried several SML codes from tasmota without success. I even not able to get some crypted or bad response from counter yet :(
Please advise how to setup this.

Regards,
Turkel.

**Comment by joba-1** [comment]
maybe the baudrate does not match?

**Reply by turkelali** [reply]
in most codes baudrate is 9600, how to find out what is baud should be ? This zenner counter not provides its docs :(

**Reply by sfromis** [reply]
Trial and error. Common ones are 4800, 9600, 19200, 38400, 57600 and 115200.

Maybe first test with the RX pin configured as a counter instead. If there is activity, the count should tick upwards. If not, the issue may be getting signals at all.

**Reply by turkelali** [reply]
All bauds are tested without luck
"Maybe first test with the RX pin configured as a counter instead"
How to do this, could you please guide?

**Reply by sfromis** [reply]
Click Configuration / Configure Module, and find Counter in the dropdown for the relevant gpio, likely gpio 3 for typical ESP8266 hardware serial.

**Reply by turkelali** [reply]
GPIO3 RX - COUNTER1 shows some values incremental 4133, 5321 ...

**Reply by turkelali** [reply]
this is also could be sensor own transmitter reflects in own sensor receiver, but I am sure I placed it correctly)

**Comment by turkelali** [comment]
```
>D
>B
=>sensor53 r
>M 1
+1,3,s,0,9600,house
```
Returns:
20:27:34.625 CMD: Serial buffer overrun
20:27:34.713 CMD: 20:27:34.028 RSL: RESULT = {"Command":"Unknown"}
20:27:34.719 RSL: RESULT = {"Command":"Unknown"}
20:27:34.966 CMD: Serial buffer overrun
